File: 1221.py
class Solution:
    def balancedStringSplit(self, s: str) -> int:
        res, current = 0, 0
        for letter in s:
            if(letter == 'L'):
                current -= 1
            else:
                current += 1
            if(current == 0):
                res += 1
        return res

File: 1221.cpp
class Solution {
public:
    int balancedStringSplit(string s) {
        int res = 0, current = 0;
        for(int i = 0; i < s.size(); i++) {
            if(s[i] == 'L') {
                current -= 1;
            } else {
                current += 1;
            }
            if(current  == 0) {
                res++;
            }
        }
        return res;
    }
};